src/components/subcomponents/join-team-button.js

```diff
@@ -1,42 +1,47 @@
 import React from "react";
 import { Button } from "@material-ui/core";
+import { joiningHackIdea } from "../../store/actions/userActions";
+import { connect } from "react-redux";
 
-export function JoinButton(props) {
+export function JoinButton({ hackDetails, user, dispatch}) {
 
   const [disable, setDisable] = React.useState(false)
 
-  const onClick = event => {
+  const joinHack = event => {
     event.preventDefault();
-    props.joinHack();
+    dispatch(joiningHackIdea(user.userId, hackDetails._id));
   };
 
   React.useEffect(() => {
-    props.team.forEach(member => {
-      console.log(member._id)
-      console.log(props.userId)
-      if(member._id === props.userId || !props.userId){
+    hackDetails.team.forEach(member => {
+      if(member._id === user.userId || !user.userId){
         setDisable(true)
```

Comment:
This shouldn't be state/effect anymore (probably never should have been), you can derive it in the render based on props...

```
return (
  <Button
    variant="outlined"
    color="primary"
    disabled={!user.userId || hackDetails.team.map(member => member._id).includes(user.userId)
    onClick={joinHack}
  >
    Join
  </Button>
)
```